lib/virt_autotest/virtual_network_utils.pm

```diff
@@ -529,23 +533,23 @@ EOF
         if ($detect_signature eq '') {
             assert_script_run("cp /etc/sysconfig/network/config /etc/sysconfig/network/config_backup");
-            $ret = assert_script_run("sed -ri \'s/^NETCONFIG_DNS_POLICY.*\$/NETCONFIG_DNS_POLICY=\"\"/g\' /etc/sysconfig/network/config");
-            $ret |= assert_script_run("echo \'#Modified by parallel_guest_migration_base module\' >> /etc/sysconfig/network/config");
+            assert_script_run("sed -ri \'s/^NETCONFIG_DNS_POLICY.*\$/NETCONFIG_DNS_POLICY=\"\"/g\' /etc/sysconfig/network/config");
+            assert_script_run("echo \'#Modified by parallel_guest_migration_base module\' >> /etc/sysconfig/network/config");
         }
         record_info("Content of /etc/sysconfig/network/config", script_output("cat /etc/sysconfig/network/config", proceed_on_failure => 1));
     }
     my $detect_signature = script_output("cat $args{resolvconf} | grep \"#Modified by parallel_guest_migration_base.pm module\"", proceed_on_failure => 1);
     my $detect_name_server = script_output("cat $args{resolvconf} | grep \"nameserver $args{resolvip}\"", proceed_on_failure => 1);
     my $detect_domain_name = script_output("cat $args{resolvconf} | grep \"$args{domainname}\"", proceed_on_failure => 1);
-    $ret |= assert_script_run("cp $args{resolvconf} /etc/resolv_backup.conf") if ($detect_signature eq '');
-    $ret |= assert_script_run("awk -v dnsvar=$args{resolvip} \'done != 1 && /^nameserver.*\$/ { print \"nameserver \"dnsvar\"\"; done=1 } 1\' $args{resolvconf} > $args{resolvconf}.tmp") if ($detect_name_server eq '');
+    assert_script_run("cp $args{resolvconf} /etc/resolv_backup.conf") if ($detect_signature eq '');
+    assert_script_run("awk -v dnsvar=$args{resolvip} \'done != 1 && /^nameserver.*\$/ { print \"nameserver \"dnsvar\"\"; done=1 } 1\' $args{resolvconf} > $args{resolvconf}.tmp") if ($detect_name_server eq '');
     if ($detect_domain_name eq '') {
-        $ret |= assert_script_run("cp $args{resolvconf} $args{resolvconf}.tmp") if (script_run("ls $args{resolvconf}.tmp") != 0);
-        $ret |= assert_script_run("sed -i -r \'s/^search/search $args{domainname}/\' $args{resolvconf}.tmp");
+        assert_script_run("cp $args{resolvconf} $args{resolvconf}.tmp") if (script_run("ls $args{resolvconf}.tmp") != 0);
+        assert_script_run("sed -i -r \'s/^search/search $args{domainname}/\' $args{resolvconf}.tmp");
     }
     if (script_run("ls $args{resolvconf}.tmp") == 0) {
-        $ret |= assert_script_run("mv $args{resolvconf}.tmp $args{resolvconf}");
-        $ret |= assert_script_run("echo \'#Modified by parallel_guest_migration_base module network $args{resolvip}\' >> $args{resolvconf}");
+        assert_script_run("mv $args{resolvconf}.tmp $args{resolvconf}");
+        assert_script_run("echo \'#Modified by parallel_guest_migration_base module network $args{resolvip}\' >> $args{resolvconf}");
```

Comment:
This subroutine needs return value, because there is such usage in automation $_ret |= config_domain_resolver(resolvip => $args{_ipaddr}, domainname => $_guest_params{dns_domainname});.

Removing return value will break something.

User can choose to die if return value is not zero. And user can also choose to continue.

So using $ret |= script_run is better.

Can you help change it back to $ret |= script_run ???